### ISSUE TITLE
NullPointerException: NavigationService handleStopService

### ISSUE THREAD

**Issue by P-Zenker** [mapbox/mapbox-navigation-android#436]
Hey,

I don't know when this error occurs, because I couldn't reproduce it on my few devices.

The error occurs for e.g. on this devices:
Android: 6.0.1 (23)
Smartphone-Model: SM-G900F (Samsung)

Android: 7.1.1 (25)
Smartphone-Model: F8331 (Sony)

Android: 6.0 (23)
Smartphone-Model: LG-H815

```
java.lang.RuntimeException: Unable to stop service com.mapbox.services.android.navigation.v5.navigation.NavigationService@9d9575e: java.lang.NullPointerException: Attempt to invoke virtual method 'double android.location.Location.getLatitude()' on a null object reference
	at android.app.ActivityThread.handleStopService(ActivityThread.java:4111)
	at android.app.ActivityThread.access$2500(ActivityThread.java:221)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1902)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:158)
	at android.app.ActivityThread.main(ActivityThread.java:7224)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'double android.location.Location.getLatitude()' on a null object reference
	at com.mapbox.services.android.navigation.v5.navigation.NavigationMetricsWrapper.rerouteEvent(NavigationMetricsWrapper.java:91)
	at com.mapbox.services.android.navigation.v5.navigation.NavigationService.sendRerouteEvent(NavigationService.java:316)
	at com.mapbox.services.android.navigation.v5.navigation.NavigationService.onDestroy(NavigationService.java:102)
	at android.app.ActivityThread.handleStopService(ActivityThread.java:4092)
	... 8 more
```
Similar error:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'double android.location.Location.getLatitude()' on a null object reference
	at com.mapbox.services.android.navigation.v5.navigation.NavigationMetricsWrapper.rerouteEvent(NavigationMetricsWrapper.java:91)
	at com.mapbox.services.android.navigation.v5.navigation.NavigationService.sendRerouteEvent(NavigationService.java:316)
	at com.mapbox.services.android.navigation.v5.navigation.NavigationService.onLocationChanged(NavigationService.java:203)
	at com.mapbox.services.android.telemetry.location.LostLocationEngine.onLocationChanged(LostLocationEngine.java:147)
	at com.mapzen.android.lost.internal.FusedLocationProviderApiImpl.reportLocation(FusedLocationProviderApiImpl.java:121)
	at com.mapzen.android.lost.internal.FusionEngine.onLocationChanged(FusionEngine.java:134)
	at android.location.LocationManager$ListenerTransport._handleMessage(LocationManager.java:297)
	at android.location.LocationManager$ListenerTransport.-wrap0(LocationManager.java)
	at android.location.LocationManager$ListenerTransport$1.handleMessage(LocationManager.java:242)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:241)
	at android.app.ActivityThread.main(ActivityThread.java:6274)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

I always check if the location is null before passing it to Mapbox inside:
- public void userOffRoute(Location location) {...}
- public void onLocationChanged(Location location) { } // LostLocationEngine or Google

Any suggestions how I can avoid this?

**Comment by Grsmto** [comment]
I think that's the same issue as #351
I'm having this issue as well, now only when I stop the navigation service it seems like.

**Comment by Grsmto** [comment]
Hey @electrostat just testing the PR that has been merged and it seems like I'm still experiencing this bug with the following stack trace when I stop the navigation:

```
11-07 11:22:58.640 6927-6927/ E/AndroidRuntime: FATAL EXCEPTION: main Process: , PID: 6927
java.lang.RuntimeException: Unable to stop service com.mapbox.services.android.navigation.v5.navigation.NavigationService@45c57db: java.lang.NullPointerException: Attempt to invoke virtual method 'double android.location.Location.getLatitude()' on a null object reference at android.app.ActivityThread.handleStopService(ActivityThread.java:3587) at android.app.ActivityThread.-wrap26(Unknown Source:0) at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1703) at android.os.Handler.dispatchMessage(Handler.java:105) at android.os.Looper.loop(Looper.java:164) at android.app.ActivityThread.main(ActivityThread.java:6541) at java.lang.reflect.Method.invoke(Native Method) at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'double android.location.Location.getLatitude()' on a null object reference
```

Thanks!

**Comment by danesfeder** [comment]
@Grsmto Do you have a way to consistently reproduce this?  Looks like we still aren't properly shutting the service down with our metrics class in some scenarios. Thanks!

**Comment by danesfeder** [comment]
Should be fixed with #511 - please let us know otherwise
